Parsers/ASimNetworkSession/Parsers/vimNetworkSessionAzureFirewall.yaml

```diff
@@ -0,0 +1,185 @@
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: srcipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dstipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dstportnumber
+    Type: int
+    Default: int(null)
+  - Name: hostname_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dvcaction
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let ICMPlookup = datatable (NetworkIcmpCode: int, NetworkIcmpType:string)
+      [
+      0,"Echo Reply",
+      1,"Unassigned",
+      2,"Unassigned",
+      3,"Destination Unreachable",
+      5,"Redirect",
+      8,"Echo",
+      9,"Router Advertisement",
+      10,"Router Solicitation",
+      11,"Time Exceeded",
+      12,"Parameter Problem",
+      13,"Timestamp",
+      14,"Timestamp Reply",
+      40,"Photuris",
+      42,"Extended Echo Request",
+      43,"Extended Echo Reply"
+      ];
+  let parser = (
+        starttime:datetime=datetime(null), 
+        endtime:datetime=datetime(null), 
+        srcipaddr_has_any_prefix:dynamic=dynamic([]), 
+        dstipaddr_has_any_prefix:dynamic=dynamic([]), 
+        dstportnumber:int=int(null), 
+        hostname_has_any:dynamic=dynamic([]), 
+        dvcaction:dynamic=dynamic([]), 
+        eventresult:string='*', 
+        disabled:bool=false)
+  {
+      let AzureFirewallNetworkRuleLogs = 
+          AzureDiagnostics
+          | where not(disabled)
+          | where Category == "AzureFirewallNetworkRule"
+          | where isnotempty(msg_s);
+      let prefilter = (T: (msg_s:string, TimeGenerated:datetime, OperationName:string)) {
+        T | where  
+        (array_length(hostname_has_any) == 0)
+         and (isnull(starttime) or TimeGenerated>=starttime) 
+         and (isnull(endtime) or TimeGenerated<=endtime) 
+         and (
+            array_length(srcipaddr_has_any_prefix)==0 
+            or has_any_ipv4_prefix(msg_s,srcipaddr_has_any_prefix)
+          ) 
+          and (
+            array_length(dstipaddr_has_any_prefix)==0 
+            or has_any_ipv4_prefix(msg_s,dstipaddr_has_any_prefix)
+          ) 
+          and (isnull(dstportnumber) or msg_s has (tostring(dstportnumber))) 
+      };
+      let AzureFirewallSessionLogs = 
+          AzureFirewallNetworkRuleLogs
+          | where OperationName in ("AzureFirewallNetworkRuleLog","AzureFirewallThreatIntelLog")
+          // -- pre-filter
+          | where (array_length(dvcaction) == 0) or (msg_s has_any (dvcaction))
+          | where (eventresult == "*") or ((eventresult == "Success") and (msg_s has "Allow")) or ((eventresult == "Failure") and (msg_s has "Deny"))
+          | invoke prefilter()
+          // -- end pre-filter
```

Comment:
would it make sense to move the pre-filtering  in to AzureFirewallNetworkRuleLogs definition?